### PR TITLE
Serialize XAttrs/ADS as ordered in protos

### PR DESCRIPTION
**Pull request by sul3n3t**
XAttrs/ADS were previously stored as a map in proto form. There is no defined ordering of maps, which can cause inconsistent binary representation of resources with multiple attrs/ads. This commit changes the types to ordered lists. The representation should be compatible.

### CODE REVIEW

**Comment by sul3n3t** [comment]
@stevvooe PTAL

File: resource.go [hunk 1]
```diff
@@ -482,7 +482,17 @@ func toProto(resource Resource) *pb.Resource {
 	}
 
 	if xattrer, ok := resource.(XAttrer); ok {
-		b.Xattr = xattrer.XAttrs()
+		// Sorts the XAttrs by name for consistent ordering.
+		keys := []string{}
+		xattrs := xattrer.XAttrs()
```
Comment (stevvooe) [inline resource.go]: Maybe, we change the XAtterer interface to return and guarantee the order?

File: proto/manifest.proto [hunk 2]
```diff
@@ -9,6 +9,8 @@ message Manifest {
 }
 
 message Resource {
+    reserved 12;
```
Comment (stevvooe) [inline proto/manifest.proto]: I don't _think_ we need to worry about breaking. The usage here is thin on xattrs. However, the types are actually equivalent.

File: proto/manifest.proto [hunk 3]
```diff
@@ -59,19 +61,19 @@ message Resource {
...
-    // TODO(stevvooe): The use of maps here may be problematic for
-    // deterministic generation. Check out this comment:
-    // https://developers.google.com/protocol-buffers/docs/proto3#backwards-compatibility
-    // Fortunately, the Go implementation correctly sorts the map keys to
-    // ensure deterministic generation, but this is not guaranteed for all
-    // implementations. If this is problem, we should generate that schema and
-    // sort by key. We can do this at any time and retain backwards
-    // compatibility.
+    // Ads stores one or more alternate data streams for the target resource.
+    map<string, bytes> ads = 13;
```
Comment (stevvooe) [inline proto/manifest.proto]: Do you mind doing this for ADS, as well?

Comment (sul3n3t) [inline proto/manifest.proto]: Sure

**Comment by stevvooe** [comment]
LGTM

These should be backwards compatible with the map type. If anyone has problems, please let us know.

**Pull request merged**